src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexerOutpointCache.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using LiteDB;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.BlockStore.AddressIndexing
+{
+    public class AddressIndexerOutpointCache
+    {
+        /// <summary>This class holds a key-value pair used to represent an item
+        /// in the LRU cache. It is necessary to maintain both values here so
+        /// that it is possible to look up items in the cache dictionary when
+        /// removing from the cache linked list.
+        /// </summary>
+        private class LRUItem
+        {
+            public LRUItem(string outPoint, OutPointData outPointData)
+            {
+                Guard.NotEmpty(outPoint, nameof(outPoint));
+                Guard.NotNull(outPointData, nameof(outPointData));
+
+                this.Key = outPoint;
+                this.Value = outPointData;
+            }
+
+            public readonly string Key;
+
+            public readonly OutPointData Value;
+        }
+
+        public const int AddressIndexOutputCacheMaxItemsDefault = 100000;
+
+        private const int BatchSize = 100;
+
+        /// <summary>The maximum number of items that can be kept in the cache until
+        /// entries will start getting evicted.</summary>
+        public readonly int MaxItems;
+
+        public int Count => this.outPointLinkedList.Count;
+
+        /// <summary>A mapping between the string representation of an outpoint and its
+        /// corresponding scriptPubKey and money value.
+        /// All access to the cache must be protected with <see cref="lockObj"/>.</summary>
+        private readonly Dictionary<string, LinkedListNode<LRUItem>> cachedOutPoints;
+
+        /// <summary>A linked list used to efficiently determine the oldest entry
+        /// in the cache. Items get added to the cache in time order and get moved to
+        /// the back of the eviction list each time they are accessed.</summary>
+        private readonly LinkedList<LRUItem> outPointLinkedList;
+
+        private object lockObj;
+
+        private readonly LiteDatabase db;
```

Comment:
no need to save to a variable, instead value passed in the constructor can be used right away without saving